src/odemis/driver/pwrcomedi.py

```diff
@@ -211,30 +209,30 @@ def getHwName(self):
         """
         return comedi.get_board_name(self._device)
 
-    def _updateIntensities(self, power, intensities):
-        for c, r, crv, intens in zip(self._channels, self._ranges, self._pwr_curve, intensities):
-            p = min(power * intens, crv[-1][1])
+    def _updateIntensities(self, power):
+        for c, r, crv, p in zip(self._channels, self._ranges, self._pwr_curve, power):
+            p = min(p, crv[-1][1])
             v = self._power_to_volt(p, crv)
             d = self._volt_to_data(v, c, r)
             logging.debug("Setting channel %d to %d = %g V = %g W", c, d, v, p)
             comedi.data_write(self._device, self._ao_subd, c, r, comedi.AREF_GROUND, d)
 
     def _updatePower(self, value):
-        self._updateIntensities(value, self.emissions.value)
-
-    def _setEmissions(self, intensities):
-        """
-        intensities (list of N floats [0..1]): intensity of each source
-        """
-        if len(intensities) != len(self._channels):
-            raise ValueError("Emission must be an array of %d floats." % len(self._channels))
-
-        # clamp intensities which cannot reach the maximum power
-        cl_intens = []
-        for crv, intens in zip(self._pwr_curve, intensities):
-            cl_intens.append(min(max(0, intens), crv[-1][1] / self.power.range[1]))
-            # TODO: if the intensity is so small that the power would be 0, force
-            # it to 0.
-
-        self._updateIntensities(self.power.value, cl_intens)
-        return cl_intens
+        self._updateIntensities(value)
+
+    # def _setEmissions(self, intensities):
+    #     """
+    #     intensities (list of N floats [0..1]): intensity of each source
+    #     """
+    #     if len(intensities) != len(self._channels):
+    #         raise ValueError("Emission must be an array of %d floats." % len(self._channels))
+    #
+    #     # clamp intensities which cannot reach the maximum power
+    #     cl_intens = []
+    #     for crv, intens in zip(self._pwr_curve, intensities):
+    #         cl_intens.append(min(max(0, intens), crv[-1][1] / self.power.range[1]))
+    #         # TODO: if the intensity is so small that the power would be 0, force
```

Comment:
How to do this?

Reply:
You'd check if the return value of `self._volt_to_data(v, c, r)` == `self._volt_to_data(crv[0][0], c, r)` (that's the minimum value).
Ideally, you'd even compute back for any value: convert the requested power (float) to the "data" (ie, int), and then convert back from the int to the actual power. This would indicate to the user the actual value set. That said, no one really cares much, excepted potentially for the "tiny power is no power" case (discussed above), where it might sometimes be useful.
Note that this wasn't handle so far, so you don't need to handle it now either.